### PR TITLE
Documentation corrections

### PR DESCRIPTION
**Pull request by nasimanousheh**
While I was reading the DIPY tutorials, I found some minor errors in grammar and spelling. Please have a look.
Thank you for all the great work!

### CODE REVIEW

**Comment by nasimanousheh** [comment]
@skoudoro have a look at your convenience.

**Review by skoudoro — APPROVED** [review]
Hi @nasimanousheh,

Overall, it looks good to me. Thank you for doing that!

it would be good if @arokem or @jhlegarreta  could have a second look.

Thanks!

**Comment by arokem** [comment]
Will do later today!

File: doc/examples/reconst_dsid.py [hunk 1]
```diff
@@ -8,8 +8,8 @@
 deconvolution in the diffusion propagator with the goal to create sharper ODFs
 with higher angular resolution.
 
-In this example we will show with simulated data how this method's ODF performs
-against standard DSI ODF and a ground truth multi tensor ODF.
+In this example we will show with simulated data how this method's ODF
+performance against standard DSI ODF and a ground truth multi tensor ODF.
```
Comment (jhlegarreta) [inline doc/examples/reconst_dsid.py:12]: I'd say that the former was correct.

File: doc/examples/segment_quickbundles.py [hunk 2]
```diff
@@ -85,10 +85,10 @@
            [  64.02451324,   88.43942261,   75.0697403 ]], dtype=float32)
 
 
-`clusters` has also attributes like `centroids` (cluster representatives), and
+`clusters` has also attributed like `centroids` (cluster representatives), and
```
Comment (jhlegarreta) [inline doc/examples/segment_quickbundles.py:88]: I think the former was correct.

File: doc/examples/streamline_length.py [hunk 3]
```diff
@@ -86,8 +86,8 @@ def simulated_bundles(no_streamlines=50, n_pts=100):
 
 """
 Often, streamlines are represented with more points than what is actually
-necessary for specific applications. Also, sometimes every streamline has
-different number of points which could be of a trouble for some algorithms.
+necessary for specific applications. Also, sometimes every streamline has a
+different number of points which could be a problem for some algorithms.
```
Comment (jhlegarreta) [inline doc/examples/streamline_length.py:90]: (...) of points, which could be (...)

File: doc/examples/streamline_length.py [hunk 4]
```diff
@@ -97,8 +97,8 @@ def simulated_bundles(no_streamlines=50, n_pts=100):
 n_pts_ds = [len(s) for s in bundle_downsampled]
 
 """
-Alternatively, the function ``approx_polygon_track`` allows to reduce the number
-of points so that they are more points in curvy regions and less points in
+Alternatively, the function ``approx_polygon_track`` allow reduces the number
```
Comment (jhlegarreta) [inline doc/examples/streamline_length.py:100]: I think "allows to reduce" is correct.

File: doc/examples/streamline_length.py [hunk 5]
```diff
@@ -139,7 +139,7 @@ def simulated_bundles(no_streamlines=50, n_pts=100):
    :align: center
 
    Initial bundle (down), downsampled at 12 equidistant points (middle),
-   downsampled not equidistantly (up).
+   downsampled not equidistant (up).
```
Comment (jhlegarreta) [inline doc/examples/streamline_length.py:142]: I think the former was correct.

File: doc/examples/streamline_tools.py [hunk 6]
```diff
@@ -80,10 +80,10 @@
 """
 The first of the tracking utilities we'll cover here is ``target``. This
 function takes a set of streamlines and a region of interest (ROI) and returns
-only those streamlines that pass though the ROI. The ROI should be an array
+only those streamlines that pass through the ROI. The ROI should be an array
 such that the voxels that belong to the ROI are ``True`` and all other voxels
 are ``False`` (this type of binary array is sometimes called a mask). This
-function can also exclude all the streamlines that pass though an ROI by
+function can also exclude all the streamlines that pass through an ROI by
```
Comment (jhlegarreta) [inline doc/examples/streamline_tools.py:86]: an ROI: a ROI

File: doc/examples/streamline_tools.py [hunk 7]
```diff
@@ -156,7 +156,7 @@
    **Corpus Callosum Sagittal**
 """
 """
-Once we've targeted on the corpus callosum ROI, we might want to find out which
+Once we've targeted th corpus callosum ROI, we might want to find out which
```
Comment (jhlegarreta) [inline doc/examples/streamline_tools.py:159]: th : the

File: doc/examples/tracking_introduction_eudx.py [hunk 8]
```diff
@@ -113,7 +113,7 @@
 stopping_criterion = ThresholdStoppingCriterion(csa_peaks.gfa, .25)
 
 """
-Again, for quality assurance we can also visualize a slice the GFA and the
+Again, for quality assurance we can also visualize slice of the GFA and the
```
Comment (jhlegarreta) [inline doc/examples/tracking_introduction_eudx.py:116]: a slice of

**Review by jhlegarreta — COMMENTED** [review]
Looks mostly good; I made a few comment in-line.

I'd squash all commits into one.

File: dipy/segment/cythonutils.pyx [hunk 9]
```diff
@@ -18,12 +18,12 @@ cdef Shape shape_from_memview(Data data) nogil:
     Parameters
     ----------
     data : memoryview object (float)
-        array for which the shape informations are retrieved
+        array for which the shape information are retrieved
```
Comment (arokem) [inline dipy/segment/cythonutils.pyx:21]:
```suggestion
        array for which the shape information is retrieved
```

File: dipy/segment/cythonutils.pyx [hunk 10]
```diff
@@ -114,19 +114,19 @@ cdef int same_shape(Shape shape1, Shape shape2) nogil:
 
 
 cdef Data2D* create_memview_2d(Py_ssize_t buffer_size, Py_ssize_t dims[MAX_NDIM]) nogil:
-    """ Create a light version of cython memory view 
+    """ Create a light version of cython memory view
```
Comment (arokem) [inline dipy/segment/cythonutils.pyx:117]:
```suggestion
    """ Create a light version of cython memory view.
```

File: doc/examples/denoise_mppca.py [hunk 11]
```diff
@@ -95,7 +95,7 @@
 patch_radius is set to 2 which corresponds to a 5x5x5 sliding window
 comprising a total of 125 voxels.
 
-To assess the performance of the Marcenko-Pastur PCA denosing algorithm,
+To assess the performance of the Marcenko-Pastur PCA denoising algorithm,
```
Comment (arokem) [inline doc/examples/denoise_mppca.py:98]: A "denosing" algorithm sounds painful 😄

File: doc/examples/denoise_mppca.py [hunk 12]
```diff
@@ -204,7 +204,7 @@
 
 In the above figure, the DKI maps obtained from the original data are shown in
 the upper panels, while the DKI maps from the denoised data are shown in the
-lower panels. Substantial improvements on measurements robustness can be
+lower panels. Substantial improvements on measurement robustness can be
```
Comment (arokem) [inline doc/examples/denoise_mppca.py:207]:
```suggestion
lower panels. Substantial improvements in measurement robustness can be
```

File: doc/examples/denoise_mppca.py [hunk 13]
```diff
@@ -244,7 +244,7 @@
 print(mean_sigma)
 
 """
-Below we use this mean noise level estimate to compute the data's nominal SNR
+Below we use this mean noise level estimate to compute the data nominal SNR
```
Comment (arokem) [inline doc/examples/denoise_mppca.py:247]:
```suggestion
Below we use this mean noise level estimate to compute the nominal SNR of the data
```

File: doc/examples/fiber_to_bundle_coherence.py [hunk 14]
```diff
@@ -19,7 +19,7 @@
    On the left this figure illustrates (in 2D) the contribution of two fiber
    points to the kernel density estimator. The kernel density estimator is the
    sum over all such locally aligned kernels. The local fiber to bundle
-   coherence shown on the right color-coded for each fiber, is obtained by
+   coherence, shown on the right color-code for each fiber, is obtained by
```
Comment (arokem) [inline doc/examples/fiber_to_bundle_coherence.py:22]:
```suggestion
   coherence, shown on the right, color-coded for each fiber, is obtained by
```

File: doc/examples/reconst_csd.py [hunk 15]
```diff
@@ -41,7 +41,7 @@
 
 """
 You can verify the b-values of the dataset by looking at the attribute
-``gtab.bvals``. Now that a datasets with multiple gradient directions is
+``gtab.bvals``. Now that datasets with multiple gradient directions are
```
Comment (arokem) [inline doc/examples/reconst_csd.py:44]:
```suggestion
``gtab.bvals``. Now that a dataset with multiple gradient directions is
```

File: doc/examples/reconst_csd.py [hunk 16]
```diff
@@ -68,12 +68,12 @@
 
 """
 Note that the ``auto_response_ssst`` function calls two functions that can be
-used separatly. First, the function ``mask_for_response_ssst`` creates a mask
-of voxels within the cuboid ROI and that meet the FA threshold constraint. This
+used separately. First, the function ``mask_for_response_ssst`` creates a mask
+of voxels within the cuboid ROI and that meets the FA threshold constraint. This
```
Comment (arokem) [inline doc/examples/reconst_csd.py:72]:
```suggestion
of voxels within the cuboid ROI that meet the FA threshold constraint. This
```

File: doc/examples/segment_quickbundles.py (hunk 2 above)
Comment (arokem) [inline doc/examples/segment_quickbundles.py:88]:
```suggestion
`clusters` also has attributes such as `centroids` (cluster representatives), and
```

File: doc/examples/streamline_formats.py [hunk 17]
```diff
@@ -87,8 +87,8 @@
 - Trk header (dict)
 - Stateful Tractogram
 
-The reason why this parameters is required is to guarantee all informations
-related to space attribute are always present.
+The reason why this parameter is required is to guarantee all information
+related to space attributes are always present.
```
Comment (arokem) [inline doc/examples/streamline_formats.py:91]:
```suggestion
related to space attributes is always present.
```

File: doc/examples/streamline_length.py (hunk 3 above)
Comment (arokem) [inline doc/examples/streamline_length.py:90]:
```suggestion
different number of points, which could be a problem for some algorithms.
```
Always a comma before "which"

File: doc/examples/streamline_length.py (hunk 5 above)
Comment (arokem) [inline doc/examples/streamline_length.py:142]: I agree with @jhlegarreta that "not equidistantly" is technically acceptable, but it's an eye sore. How about the following? 

```suggestion
   downsampled with points that are not equidistant (up).
```

File: doc/examples/streamline_tools.py (hunk 6 above)
Comment (arokem) [reply doc/examples/streamline_tools.py:86]: This is a tricky one. I pronounce this as "ar-oh-eye", spelling out each letter, which means that I would write "an ROI", but I know that others pronounce this like the name "Roy", which would have you with "a ROI". I say po-tay-toh, you say po-tah-to. I think it's fine as is.

File: doc/examples/streamline_tools.py (hunk 7 above)
Comment (arokem) [inline doc/examples/streamline_tools.py:159]:
```suggestion
Once we've targeted the corpus callosum ROI, we might want to find out which
```

**Review by arokem — COMMENTED** [review]
Thanks for doing this! I had a few small comments/suggestions.

File: doc/examples/reconst_dsid.py (hunk 1 above)
Comment (skoudoro) [inline doc/examples/reconst_dsid.py:12]:
```suggestion
performs against standard DSI ODF and a ground truth multi tensor ODF.
```

File: doc/examples/streamline_length.py (hunk 4 above)
Comment (skoudoro) [inline doc/examples/streamline_length.py:100]:
```suggestion
Alternatively, the function ``approx_polygon_track`` allows reducing the number
```

File: doc/examples/tracking_introduction_eudx.py (hunk 8 above)
Comment (skoudoro) [inline doc/examples/tracking_introduction_eudx.py:116]:
```suggestion
Again, for quality assurance, we can also visualize a slice of the GFA and the
```

**Comment by skoudoro** [comment]
Thank you @nasimanousheh for this cleaning. All comments have been addressed.  merging

**Comment by skoudoro** [comment]
Thanks @arokem and @jhlegarreta  for the review also!

**Pull request merged**